### PR TITLE
docs: documentation for account compression program

### PR DESCRIPTION
**Pull request by anoushk1234**
I've added documentation for the account compression program. Here's a summary of the changes.

I've added 3 pages:
- overview
- usage
- core concepts

Overview contains brief on the program, motivation, background, packages and guide to run the tests.
Usage contains examples to use the TS SDK
Core concepts contains a concise explanation of the core technology used in account compression and how it ties into compressing NFTs with easy to understand examples.

### CODE REVIEW

**Comment by anoushk1234** [comment]
> TS usage / examples look good. Explanation of algorithm is excellent, but a little bit off. Will highlight particular lines
> 
> Can you explain what the `package.json` changes are for?

Docusaurus wasn't serving the page for me on localhost 3000 without specifying the host and port on the start command, after that it did serve the page but gave an error saying switchConfig is undefined which was fixed after updating the packages

**Comment by ngundotra** [comment]
Please undo your package json changes

**Comment by ngundotra** [comment]
Otherwise, this looks good. Will need some touch up after merging, but looks good to me. Great work!

Can you remove the digital asset examples? Those are bound to be out of date soon

**Comment by anoushk1234** [comment]
> Otherwise, this looks good. Will need some touch up after merging, but looks good to me. Great work!
> 
> Can you remove the digital asset examples? Those are bound to be out of date soon

Thanks! Made both the changes, glad I could contribute and thanks for coordinating, looking forward to doing more contributions

**Comment by ngundotra** [comment]
Looks good to me. cc @joncinque for help with reviewing changes to Docusaurus

File: docs/sidebars.js
Comment (joncinque) [inline docs/sidebars.js:28]: nit: can this be moved to the end, after the confidential extension?

**Review by joncinque — COMMENTED** [review]
The rest looks good to me, just one little nit to move the section down

File: docs/sidebars.js
Comment (ngundotra) [reply docs/sidebars.js]: Fixed!

**Review by joncinque — APPROVED** [review]
Looks good to me! I'll let you merge it

**Pull request merged**
